geoutils/pointcloud/pointcloud.py

```diff
@@ -1 +1,393 @@
+
+        # Check shape
+        if array.shape[0] != 3 and array.shape[1] != 3:
+            raise ValueError("Array must be of shape 3xN or Nx3.")
+
+        # Make the first axis the one with size 3
+        if array.shape[0] != 3:
+            array_in = array.T
+        else:
+            array_in = array
+
+        # Build geodataframe
+        gdf = gpd.GeoDataFrame(
+            geometry=gpd.points_from_xy(x=array_in[0, :], y=array_in[1, :], crs=crs), data={data_column: array_in[2, :]}
+        )
+
+        return cls(filename_or_dataset=gdf, data_column=data_column)
+
+    @classmethod
+    def from_tuples(
+        cls, tuples_xyz: Iterable[tuple[Number, Number, Number]], crs: CRS, data_column: str = "z"
+    ) -> PointCloud:
+        """Create point cloud from an iterable of 3-tuples (X coordinate, Y coordinate, Z value)."""
+
+        return cls.from_array(np.array(tuples_xyz), crs=crs, data_column=data_column)
+
+    @classmethod
+    def from_xyz(cls, x: ArrayLike, y: ArrayLike, z: ArrayLike, crs: CRS, data_column: str = "z") -> PointCloud:
+        """Create point cloud from three 1D array-like coordinates for X/Y/Z."""
+
+        return cls.from_array(np.stack((x, y, z)), crs=crs, data_column=data_column)
+
+    def to_array(self) -> NDArrayNum:
+        """Convert point cloud to a 3 x N array of X coordinates, Y coordinates and Z values."""
+
+        return np.stack((self.geometry.x.values, self.geometry.y.values, self.ds[self.data_column].values), axis=0)
+
+    def to_tuples(self) -> Iterable[tuple[Number, Number, Number]]:
+        """Convert point cloud to a list of 3-tuples (X coordinate, Y coordinate, Z value)."""
+
+        return list(zip(self.geometry.x.values, self.geometry.y.values, self.ds[self.data_column].values))
+
+    def to_xyz(self) -> tuple[NDArrayNum, NDArrayNum, NDArrayNum]:
+        """Convert point cloud to three 1D arrays of coordinates for X/Y/Z."""
+
+        return self.geometry.x.values, self.geometry.y.values, self.ds[self.data_column].values
+
+    def pointcloud_equal(self, other: PointCloud, **kwargs: Any) -> bool:
+        """
+        Check if two point clouds are equal.
+
+        This means that:
+        - The two vectors (geodataframes) are equal.
+        - The data column is the same for both point clouds.
+
+        Keyword arguments are passed to geopandas.assert_geodataframe_equal.
+        """
+
+        # Vector equality
+        vector_eq = super().vector_equal(other, **kwargs)
+        # Data column equality
+        data_column_eq = self.data_column == other.data_column
+
+        return all([vector_eq, data_column_eq])
+
+    def grid(
+        self,
+        ref: gu.Raster | None,
+        grid_coords: tuple[NDArrayNum, NDArrayNum] | None,
+        resampling: Literal["nearest", "linear", "cubic"],
+        dist_nodata_pixel: float = 1.0,
+    ) -> gu.Raster:
+        """Grid point cloud into a raster."""
```

Comment:
I think the arguments should be defined, no?

Reply:
Done.